bioblend/_tests/TestGalaxyJobs.py

```diff
@@ -74,7 +76,38 @@ def test_get_jobs(self):
         jobs = self.gi.jobs.get_jobs(date_range_min=tomorrow.strftime('%Y-%m-%d'))
         self.assertEqual(len(jobs), 0)
         jobs = self.gi.jobs.get_jobs(date_range_min=datetime.today().strftime('%Y-%m-%d'), history_id=self.history_id)
-        self.assertEqual(len(jobs), 3)
+        self.assertGreaterEqual(len(jobs), 3)
+
+    @test_util.skip_unless_galaxy('release_21.05')
+    def test_get_jobs_with_filtering(self):
+        dataset = {'src': 'hda', 'id': self.dataset_id}
+        invocation1 = self.gi.workflows.invoke_workflow(
+            self.workflow_id,
+            inputs={'Input 1': dataset, 'Input 2': dataset},
+            history_id=self.history_id,
+            inputs_by='name',
+        )
+        invocation2 = self.gi.workflows.invoke_workflow(
+            self.workflow_id,
+            inputs={'Input 1': dataset, 'Input 2': dataset},
+            history_id=self.history_id,
+            inputs_by='name',
+        )
+        self.gi.invocations.wait_for_invocation(invocation1['id'])
+        self.gi.invocations.wait_for_invocation(invocation2['id'])
+
+        jobs = self.gi.jobs.get_jobs(history_id=self.history_id)
+        self.assertGreaterEqual(len(jobs), 2)
+        job_id_lim = jobs[1]['id']
+        jobs = self.gi.jobs.get_jobs(history_id=self.history_id, limit=1, offset=1)
+        self.assertEqual(len(jobs), 1)
+        self.assertEqual(jobs[0]['id'], job_id_lim)
+        jobs = self.gi.jobs.get_jobs(invocation_id=invocation1['id'])
+        self.assertEqual(len(jobs), 1)
+        job_id_inv = jobs[0]['id']
+        jobs = self.gi.jobs.get_jobs(workflow_id=invocation1['workflow_id'])
```

Comment:
```suggestion
        jobs = self.gi.jobs.get_jobs(workflow_id=self.workflow_id)
```
This makes it pass for me. But I've no idea why, I'd have thought `self.workflow_id` and `invocation1['id']` should be the same...

Reply:
It really seems to be different, and the value of `invocation1['workflow_id']` returns `No such workflow found`. I guess this is a bug introduced since my commit that @rikeshi linked (since it is apparently passing there).

Reply:
Thanks, but this change does not make it pass for me. Still get an empty list of jobs. Are you running galaxy dev with the latest updates?

Reply:
```suggestion
        jobs = self.gi.jobs.get_jobs(workflow_id=workflow_id)
```

Reply:
@rikeshi Can you try with https://github.com/nsoranzo/galaxy/commit/4fb10d2b0761875f2a9f4bb6bd9d378e47d5fb1f ?

Reply:
I've opened https://github.com/galaxyproject/galaxy/pull/11751 , which should fix this.

Reply:
It does indeed fix it!